cupy/cuda/cub.pyx

```diff
@@ -200,6 +204,58 @@ def device_segmented_reduce(ndarray x, op, axis, out=None,
     return y
 
 
+def device_csrmv(int n_rows, int n_cols, int nnz, ndarray values,
+                 ndarray indptr, ndarray indices, ndarray x):
+    cdef ndarray y, ws
+    cdef void* values_ptr
+    cdef void* row_offsets_ptr
+    cdef void* col_indices_ptr
+    cdef void* x_ptr
+    cdef void* y_ptr
+    cdef void* ws_ptr
+    cdef int dtype_id
+    cdef size_t ws_size
+    cdef Stream_t s
+
+    if x.ndim != 1:
+        raise ValueError('array must be 1d')
+    if x.size != n_cols:
+        raise ValueError("size of array does not match the CSR matrix")
+
+    if values.dtype == x.dtype:
+        dtype = values.dtype
+    else:
+        dtype = numpy.promote_types(values.dtype, x.dtype)
+        values = values.astype(dtype, "C", None, None, False)
+        x = x.astype(dtype, "C", None, None, False)
+
+    # CSR matrix attributes
+    values_ptr = <void*>values.data.ptr
+    row_offsets_ptr = <void*>indptr.data.ptr
+    col_indices_ptr = <void*>indices.data.ptr
+
+    x_ptr = <void*>x.data.ptr
+
+    # prepare output array
+    y = ndarray((n_rows,), dtype=dtype)
+    y_ptr = <void*>y.data.ptr
+
+    s = <Stream_t>stream.get_current_stream_ptr()
+    dtype_id = _get_dtype_id(dtype)
+
+    # get workspace size and then fire up
+    ws_size = cub_device_spmv_get_workspace_size(
+        values_ptr, row_offsets_ptr, col_indices_ptr, x_ptr, y_ptr, n_rows,
+        n_cols, nnz, s, dtype_id)
+    ws = ndarray(ws_size, numpy.int8)
+    ws_ptr = <void*>ws.data.ptr
```

Comment:
Just a nitpick: I noticed that in other modules like cufft and cudnn, we simply allocate raw buffers for temporary workspaces. I already added a fix for other CUB functions in #2682, could you do it here too?
```python
    ws = memory.alloc(ws_size)
    ws_ptr = <void*>ws.ptr
```
You may need to `cimport` the `memory` module depending on which PR is merged first 😛

Reply:
#2682 is merged, so `memory` is imported and we just need to change these two lines.